### PR TITLE
Integrate Paths class into qTox code

### PR DESCRIPTION
**Pull request by sudden6**
- [ ] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

This PR provides the integration for the Paths module in the existing code. I decided to pass `Paths` via a getter function through the `Settings` class in order to keep the amount of changes manageable.

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/5440)

### CODE REVIEW

File: src/persistence/profile.cpp
Comment (Diadlo) [inline src/persistence/profile.cpp]: const reerence

File: src/persistence/profile.cpp
Comment (Diadlo) [inline src/persistence/profile.cpp:103]: I think it's better to add ` + ".ini"` here and reimplement `getDbPath` using `Settings::getInstance().getPaths()` (i.e. without `getProfilePath`) and move it closer to each other

File: src/persistence/settings.cpp
Comment (Diadlo) [inline src/persistence/settings.cpp]: Why not?
```
settings = new Settings{paths};
return settings;
```

File: src/persistence/settings.cpp
Comment (Diadlo) [inline src/persistence/settings.cpp]: Could comment be moved on the previous line (before `notifySound`) to avoid a lot of line breaks?

File: src/persistence/settings.cpp
Comment (Diadlo) [inline src/persistence/settings.cpp:728]: Maybe move this logic in `Paths`?

File: src/widget/form/chatform.cpp
Comment (Diadlo) [inline src/widget/form/chatform.cpp]: `+=`?

File: src/persistence/profile.cpp
Comment (Diadlo) [inline src/persistence/profile.cpp:108]: And maybe move in `Paths`? Also used in `ProfileImporter`

File: src/persistence/settings.cpp
Comment (Diadlo) [inline src/persistence/settings.cpp:786]: Portable mode now removed?

**Review by Diadlo — CHANGES_REQUESTED** [review]

File: src/persistence/profile.cpp
Comment (sudden6) [reply src/persistence/profile.cpp:103]: I'd prefer to do this later to keep the number of changes under control here.

File: src/persistence/profile.cpp
Comment (sudden6) [reply src/persistence/profile.cpp:108]: Same

File: src/persistence/settings.cpp
Comment (sudden6) [reply src/persistence/settings.cpp:728]: Same

File: src/persistence/settings.cpp
Comment (sudden6) [reply src/persistence/settings.cpp:786]: Moved to `Paths` and changed a bit. Current portable mode is horribly broken because it always writes to the user directory.

File: src/persistence/settings.cpp
Comment (sudden6) [reply src/persistence/settings.cpp]: done

File: src/persistence/settings.cpp
Comment (sudden6) [reply src/persistence/settings.cpp]: Done

File: src/widget/form/chatform.cpp
Comment (sudden6) [reply src/widget/form/chatform.cpp]: Done

File: src/persistence/profile.cpp
Comment (sudden6) [reply src/persistence/profile.cpp]: Done

File: src/persistence/profile.cpp
Comment (Diadlo) [reply src/persistence/profile.cpp:103]: Ok. But both methods are already touched

**Review by Diadlo — CHANGES_REQUESTED** [review]
> Reviewed 10 of 14 files at r1, 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5440)* status: 0 of 1 LGTMs obtained

___
*[src/persistence/settings.cpp, line 788 at r1](https://reviewable.io/reviews/qtox/qtox/5440#-LRgZksI98cmRyHe4sYy-r1-788:-LRpfSAH3tQ-w5PyTRbJ:bhl5nck) ([raw file](https://github.com/qtox/qtox/blob/8ab3a449e1377ec0039616cd7696c12dd7d3be14/src/persistence/settings.cpp#L788)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

Moved to `Paths` and changed a bit. Current portable mode is horribly broken because it always writes to the user directory.
</blockquote></details>

But nothing is passed in `Paths` constructor. So it's impossible to run qTox portable

**Review by sudden6 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5440)* status: 0 of 1 LGTMs obtained

___
*[src/persistence/settings.cpp, line 788 at r1](https://reviewable.io/reviews/qtox/qtox/5440#-LRgZksI98cmRyHe4sYy-r1-788:-LRqAOt33H6eYD_qtX6U:b-2nwfjx) ([raw file](https://github.com/qtox/qtox/blob/8ab3a449e1377ec0039616cd7696c12dd7d3be14/src/persistence/settings.cpp#L788)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

But nothing is passed in `Paths` constructor. So it's impossible to run qTox
</blockquote></details>

qTox will automatically run in portable mode if it finds a `qtox.ini` file besides the executable

In the future I want to add a command line switch to start in portable mode

**Review by Diadlo — APPROVED** [review]
:lgtm_strong:

> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5440#-:-LRu71Kj0Z4XXk7b6ijd:b-jbsf8z)* status: :shipit: complete! 1 of 1 LGTMs obtained

**Comment by anthonybilinski** [comment]
I've pushed the upstream branch `path_rework` so that we can merge this change before the backwards compatible upgrade part is done, unblocking @sphaerophoria's multiple PRs that also touch paths. Once your upgrade code is done, we can merge `path_rework` back into `master`, so that `master` remains in a working state, as we discussed a couple days ago, @sudden6. I've set the base branch of this PR to `path_rework`, which causes change in diff since it is currently equal to `master`.

**Review by anthonybilinski — APPROVED** [review]
:lgtm_strong:

> Reviewed 10 of 14 files at r1, 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5440#-:-LSwqIa16t8lw0VA048N:b-jbsf8z)* status: :shipit: complete! 2 of 1 LGTMs obtained

___
*[src/persistence/profile.cpp, line 108 at r2](https://reviewable.io/reviews/qtox/qtox/5440#-LSwloWu0erQA29NOypO:-LSwloWu0erQA29NOypP:b-wumxoy) ([raw file](https://github.com/qtox/qtox/blob/80ee2bc68b1a6aac13e35c4b47f2b7cfb3bd7fdd/src/persistence/profile.cpp#L108)):*
> ```cpp
> QString Profile::getToxSavePath(const QString &name)
> {
>     return Settings::getInstance().getPaths().getToxSaveDir() % name % ".tox";
> ```

I agree with Diadlo here. It's confusing that getProfilePath and getToxSavePath act differently. I think getProfilePath should also include the extension. It's only called in 4 places (plus one special place where we get db path but call getProfilePath) per my index, so doesn't seem like it would cause a lot of new changes. If you disagree still feel free to merge, I won't block the PR on this.

**Comment by anthonybilinski** [comment]
@sudden6 don't forget to merge to `path_rework` instead of `master` :)

**Comment by sudden6** [comment]
@anthonybilinski done

**Pull request closed without merging**
